### PR TITLE
[guides/images] Streamline (old) image integration content

### PR DESCRIPTION
**Pull request by sarah11918**
This PR removes content about the Image integration `@astrojs/image` from the regular guide advice. In its place is a deprecation notice and a link to the package README for full documentation of this integration.

Additionally, the note at the top of the page re: experimental assets is updated to better reflect its impending un-experimentalness, and using it now is encouraged!

New top of the page:
Screenshot from 2023-07-12 18-07-10

New section on the image integration:
Screenshot from 2023-07-12 18-08-16

### CODE REVIEW

File: src/content/docs/en/guides/images.mdx [hunk 1]
```diff
@@ -127,242 +129,24 @@ This will require you to update your existing images, remove the current Astro i
...
-
-It will be treated as a remote image, which requires either both [`width`](/en/guides/integrations-guide/image/#width) and [`height`](/en/guides/integrations-guide/image/#height), or one dimension and an [`aspectRatio`](/en/guides/integrations-guide/image/#aspectratio) attribute.
-
-A value for the `format` attribute (e.g. png, avif) to transform your image is required.
-
-[All other properties](/en/guides/integrations-guide/image/#image-) are optional.
-
-Your original image will be copied unprocessed to the build folder, like all files located in `public/`, and Astro's image integration will also return optimized versions of the image.
+:::note[to be deprecated in v3.0]
+This integration will no longer be actively supported in Astro v3.0.
```
Comment (Princesseuh) [inline src/content/docs/en/guides/images.mdx:133]: Maybe `@astrojs/image` should be included in this paragraph so people know what integration we're talking about?

File: src/content/docs/en/guides/images.mdx [hunk 2]
```diff
@@ -8,10 +8,12 @@ import Since from '~/components/Since.astro';
 
 Astro provides several ways for you to use images on your site, whether they are stored locally inside your project, linked to remotely, or stored in a CMS or CDN!
 
-:::note[using Assets (Experimental)]
-Experimental asset support was added in `astro@2.1.0`. If you have enabled assets support, see the [Assets (Experimental) Guide](/en/guides/assets/) for extra information.
+:::note[`astro:assets` (Experimental - coming in v3.0)]
+The experimental `astro:assets` module will be built in to `astro@3.0`.
```
Comment (Princesseuh) [inline src/content/docs/en/guides/images.mdx:12]: Experimental assets is technically already "built-in", it's just under a flag. Maybe "enabled by default" or something similar would be better?

Comment (sarah11918) [inline src/content/docs/en/guides/images.mdx:12]:
```suggestion
The experimental `astro:assets` module will be enabled by default in `astro@3.0`.
```

**Review by Yan-Thomas — APPROVED** [review]
LGTM!

**Pull request merged**